### PR TITLE
Move publish message creation to the super class

### PR DESCRIPTION
**Pull request by dabonnie**
* Add interface to retrieve metric name
* Include units in memory metric name

Signed-off-by: Devin Bonnie <dbbonnie@amazon.com>

### CODE REVIEW

File: system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.hpp [hunk 1]
```diff
@@ -85,6 +85,8 @@ class PeriodicMeasurementNode : public system_metrics_collector::Collector,
    */
   virtual void performPeriodicMeasurement();
 
+  void publishStatisticMessage() override;
```
Comment (mm318) [inline system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.hpp]: Add doc comment on this member function declaration.

File: system_metrics_collector/src/system_metrics_collector/metrics_message_publisher.hpp [hunk 2]
```diff
@@ -55,8 +55,13 @@ class MetricsMessagePublisher
    * ROS2 timer per the publish_period)
    */
   virtual void publishStatisticMessage() = 0;
-};
 
+  /**
+   * Return the name to use for this metric
+   * @return a string of the name for this measured metric
+   */
+  virtual std::string getMetricName() = 0;
```
Comment (thomas-moulard) [inline system_metrics_collector/src/system_metrics_collector/metrics_message_publisher.hpp]:
```suggestion
  virtual std::string getMetricName() const = 0;
```

File: system_metrics_collector/src/system_metrics_collector/linux_cpu_measurement_node.cpp [hunk 3]
```diff
@@ -116,11 +116,9 @@ system_metrics_collector::ProcCpuData LinuxCpuMeasurementNode::makeSingleMeasure
   }
 }
 
-void LinuxCpuMeasurementNode::publishStatisticMessage()
+std::string LinuxCpuMeasurementNode::getMetricName()
```
Comment (thomas-moulard) [inline system_metrics_collector/src/system_metrics_collector/linux_cpu_measurement_node.cpp]:
```suggestion
std::string LinuxCpuMeasurementNode::getMetricName() const
```

File: system_metrics_collector/src/system_metrics_collector/linux_cpu_measurement_node.hpp [hunk 4]
```diff
@@ -85,9 +85,10 @@ class LinuxCpuMeasurementNode : public system_metrics_collector::PeriodicMeasure
   virtual system_metrics_collector::ProcCpuData makeSingleMeasurement();
 
   /**
-   * Publish the statistics derived from the collected measurements
+   * Return the name to use for this metric
+   * @return a string of the name for this measured metric
    */
-  void publishStatisticMessage() override;
+  std::string getMetricName() override;
```
Comment (thomas-moulard) [inline system_metrics_collector/src/system_metrics_collector/linux_cpu_measurement_node.hpp]:
```suggestion
  std::string getMetricName() const override;
```

File: system_metrics_collector/src/system_metrics_collector/linux_memory_measurement_node.cpp [hunk 5]
```diff
@@ -53,11 +53,10 @@ double LinuxMemoryMeasurementNode::periodicMeasurement()
   return processMemInfoLines(read_string);
 }
 
-void LinuxMemoryMeasurementNode::publishStatisticMessage()
+std::string LinuxMemoryMeasurementNode::getMetricName()
```
Comment (thomas-moulard) [inline system_metrics_collector/src/system_metrics_collector/linux_memory_measurement_node.cpp]:
```suggestion
std::string LinuxMemoryMeasurementNode::getMetricName() const
```

File: system_metrics_collector/src/system_metrics_collector/linux_memory_measurement_node.hpp [hunk 6]
```diff
@@ -57,9 +57,10 @@ class LinuxMemoryMeasurementNode : public system_metrics_collector::PeriodicMeas
   double periodicMeasurement() override;
 
   /**
-   * Publish the statistics derived from the collected measurements
+   * Return the name to use for this metric
+   * @return a string of the name for this measured metric
    */
-  void publishStatisticMessage() override;
+  std::string getMetricName() override;
```
Comment (thomas-moulard) [inline system_metrics_collector/src/system_metrics_collector/linux_memory_measurement_node.hpp]:
```suggestion
  std::string getMetricName() const override;
```

File: system_metrics_collector/test/system_metrics_collector/test_periodic_measurement_node.cpp [hunk 7]
```diff
@@ -65,6 +66,11 @@ class TestPeriodicMeasurementNode : public ::system_metrics_collector::PeriodicM
 
   void publishStatisticMessage() override {}
 
+  std::string getMetricName()
```
Comment (thomas-moulard) [inline system_metrics_collector/test/system_metrics_collector/test_periodic_measurement_node.cpp]:
```suggestion
  std::string getMetricName() const
```

**Pull request merged**